tools/make_patch.c

```diff
@@ -0,0 +1,437 @@
+	const struct Symbol *symbol = symbol_find(symbols, sym_name);
+	return (absolute ? symbol->offset : symbol->address) + offset_mod;
+}
+
+void interpret_command(char *command, const struct Symbol *current_hook, const struct Symbol *symbols, struct Buffer *patches, FILE *restrict new_rom, FILE *restrict orig_rom, FILE *restrict output) {
+	// Strip all leading spaces and all but one trailing space
+	int x = 0;
+	for (int i = 0; command[i]; i++) {
+		if (!isspace((unsigned char)command[i]) || (i > 0 && !isspace((unsigned char)command[i - 1]))) {
+			command[x++] = command[i];
+		}
+	}
+	command[x - (x > 0 && isspace((unsigned char)command[x - 1]))] = '\0';
+
+	// Count the arguments
+	int argc = 0;
+	for (const char *c = command; *c; c++) {
+		if (isspace((unsigned char)*c)) {
+			argc++;
+		}
+	}
+
+	// Get the arguments
+	char *argv[argc]; // VLA
+	char *arg = command;
+	for (int i = 0; i < argc; i++) {
+		for (; *arg && !isspace((unsigned char)*arg); arg++);
+		if (!*arg) {
+			break;
+		}
+		*arg++ = '\0';
+		argv[i] = arg;
+	}
+
+	// Use the arguments
+	if (!strcmp(command, "patch") || !strcmp(command, "PATCH") || !strcmp(command, "patch_") || !strcmp(command, "PATCH_")) {
+		if (!current_hook) {
+			error_exit("Error: No current patch for command: \"%s\"", command);
+		}
+		int current_offset = current_hook->offset;
+		if (argc > 0) {
+			current_offset += parse_number(argv[0], 0);
+		}
+		if (fseek(orig_rom, current_offset, SEEK_SET)) {
+			error_exit("Error: Cannot seek to \"vc_patch %s\" in the original ROM\n", current_hook->name);
+		}
+		if (fseek(new_rom, current_offset, SEEK_SET)) {
+			error_exit("Error: Cannot seek to \"vc_patch %s\" in the new ROM\n", current_hook->name);
+		}
+		const struct Symbol *current_hook_end = symbol_find_cat(symbols, current_hook->name, "_End");
+		int length = current_hook_end->offset - current_offset;
+		buffer_append(patches, &(struct Patch){current_offset, length});
+		bool modified = false;
+		if (length == 1) {
+			int c = getc(new_rom);
+			modified = c != getc(orig_rom);
+			fprintf(output, isupper((unsigned char)command[0]) ? "0x%02X" : "0x%02x", c);
+		} else {
+			fprintf(output, command[strlen(command) - 1] == '_' ? "a%d: " : "a%d:", length);
+			for (int i = 0; i < length; i++) {
+				if (i) {
+					putc(' ', output);
+				}
+				int c = getc(new_rom);
+				modified |= c != getc(orig_rom);
+				fprintf(output, isupper((unsigned char)command[0]) ? "%02X" : "%02x", c);
+			}
+		}
+		if (!modified) {
+			fprintf(stderr, PROGRAM_NAME ": Warning: \"vc_patch %s\" doesn't alter the ROM\n", current_hook->name);
+		}
+
+	} else if (!strcmp(command, "dws") || !strcmp(command, "DWS") || !strcmp(command, "dws_") || !strcmp(command, "DWS_")) {
+		if (argc < 1) {
+			error_exit("Error: Invalid arguments for command: \"%s\"", command);
+		}
+		fprintf(output, command[strlen(command) - 1] == '_' ? "a%d: " : "a%d:", argc * 2);
```

Comment:
split this into an `if (strchr(command, '_')) putc(' ', output);`

Reply:
All three commands which need an option space for matching just put it as an inline modification to the statement they'd otherwise have:

```c
fprintf(output, /*command[strlen(command) - 1] == '_' ? "a%d: " :*/ "a%d:", length);

fprintf(output, /*command[strlen(command) - 1] == '_' ? "a%d: " :*/ "a%d:", argc * 2);

fputs(/*command[strlen(command) - 1] == '_' ? "a1: " :*/ "a1:", output);
```

Reply:
I meant mostly:
```c
fprintf(output, "a%d:", length);
if (strchr(command, '_')) putc(' ', output);
```
Mostly because the original is so long, but it's fine I guess.